tools/okparser/src/parser.py

```diff
@@ -0,0 +1,85 @@
+import re
+from pathlib import Path
+
+import requests
+import typer
+import yaml
+
+from tools.okparser.src import utils
+
+__REQUIRED_FIELDS__ = frozenset({"bom", "tool-list"})
+
+
+def get_wiki_data(items: list[str]) -> list[dict[str, str]]:
+    """search the wikidata api for items.
+
+    Args:
+        items: list of items(bom items, tools) to be searched.
+    Returns:
+        A list of search descriptions for each item.
+    """
+    item_list = []
+    with typer.progressbar(items, color=True) as progress:
+        for item in progress:
+            item_dict = {"identifier": "", "description": item.strip(), "link": ""}
+            # look for 'or' keyword in text and search each item
+            sub_items = item.split("or")
+            for sub_item in sub_items:
+                response = requests.get(utils.get_url(sub_item)).json()
```

Comment:
This code needs to be much more interactive, at least in later iterations. A few things I can think of

* If there are zero results, the user should have the ability to enter an alternative for the search string
* breaking the free text input on commas and the word 'or' is a great start, but user should be able to add/remove atoms from BOM and tool list
* user may want to choose something other than the first search result

Reply:
This makes a lot of sense. Perhaps we can push all the interactiveness into a later PR. Will raise that in the next meeting as well